File: ProblemSet_30xx/Problem_3074/solution.py
import heapq
from typing import List


class Solution:
  def minimumBoxes(self, apple: List[int], capacity: List[int]) -> int:
    result = 0
    apple_count = sum(apple)
    inversed_capacity = [-value for value in capacity]
    heapq.heapify(inversed_capacity)
    
    while len(inversed_capacity) > 0 and apple_count > 0:
      capac = heapq.heappop(inversed_capacity)
      apple_count += capac
      result += 1
    return result
      

File: ProblemSet_30xx/Problem_3074/solution.cs
Solution s = new();
Console.WriteLine(s.MinimumBoxes([1,3,2], [4,3,1,5,2]));


public class Solution {
    public int MinimumBoxes(int[] apple, int[] capacity) {
        int sum = apple.Sum();
        Array.Sort(capacity);
        int i;
        for (i = capacity.Length - 1; sum > 0 && i >= 0; i--)
        {
            sum -= capacity[i];
        }
        return capacity.Length - 1 - i;
    }
}